packages/frontend/core/src/components/attachment-viewer/utils.ts

```diff
@@ -0,0 +1,54 @@
+import type { AttachmentBlockModel } from '@blocksuite/affine/blocks';
+
+import { downloadBlob } from '../../utils/resource';
+
+export async function getAttachmentBlob(model: AttachmentBlockModel) {
+  const sourceId = model.sourceId;
+  if (!sourceId) {
+    return null;
+  }
+
+  const doc = model.doc;
+  let blob = await doc.blobSync.get(sourceId);
+
+  if (blob) {
+    blob = new Blob([blob], { type: model.type });
+  }
+
+  return blob;
+}
+
+export async function download(model: AttachmentBlockModel) {
+  const blob = await getAttachmentBlob(model);
+  if (!blob) return;
+
+  await downloadBlob(blob, model.name);
+}
+
+export function renderItem(
+  scroller: HTMLElement | null,
+  id: number,
+  imageData: ImageData
+) {
+  if (!scroller) return;
+
+  const wrapper = scroller.querySelector(`[data-index="${id}"]`);
```

Comment:
this selector is quite normal. make it more unique to avoid unexpected element queried